core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -386,7 +387,7 @@ public final void runInContext() {
             }
           } catch (Throwable t) {
             stream.cancel(Status.CANCELLED.withCause(t).withDescription("Failed to read message."));
-            return;
+            Throwables.propagateIfPossible(t);
```

Comment:
Is this listener supposed to log the exception in the current JVM in some way or another? (I'm not familiar with this part of the code.)

I still find catching and not rethrowing every Throwable suspect, particularly Errors, e.g., `VirtualMachineError`s like OOM. These errors are severe enough that default behavior seems preferable. Things are often messed up enough that handling code may end up failing while trying to do something else with the result that the exception is mangled or lost.

Anything that keeps NPEs from being logged in the current JVM seems like a bad thing, though it's a `RuntimeException`, not an `Error`.